datacapturing/src/main/java/de/cyface/datacapturing/backend/DataCapturingBackgroundService.java

```diff
@@ -253,25 +261,26 @@ private void init() {
      * @param data The data to send appended to this message. This may be <code>null</code> if no data needs to be send.
      */
     void informCaller(final int messageCode, final Parcelable data) {
-        Message msg = Message.obtain(null, messageCode);
+        final Message msg = Message.obtain(null, messageCode);
 
         if (data != null) {
-            Bundle dataBundle = new Bundle();
+            final Bundle dataBundle = new Bundle();
             dataBundle.putParcelable("data", data);
             msg.setData(dataBundle);
         }
 
         if (BuildConfig.DEBUG)
             Log.v(TAG, String.format("Sending message %d to %d callers.", messageCode, clients.size()));
-        Set<Messenger> iterClients = new HashSet<>(clients);
```

Comment:
Zu dem FIXME (das ich noch beheben muss vor dem merge):

Ich verstehe nicht ganz, wieso wir eine Kopie der List zum Iterieren anlegen. Das würde Sinn ergeben, wenn wir die originale Liste nicht modifizieren wollen, hier modifizieren wir die aber ohnehin ... ?

Reply:
Man kann nicht gleichzeitig über eine Collection iterieren und Element entfernen. Deswegen muss die collection für die Iteration kopiert werden.

Reply:
Zu dem Zeitpunkt noch nicht - da wollte ich erst fragen bevor ich etwas zerstöre ;-) Aber in dem protcol-orientated programming Video von dir bei ~8:10 min war ein Hinweis der mich hieran erinnert hat.

Reply:
Hätte gedacht, dass das Java einem abnimmt: Quasi nehme dir ein Objekt aus der Collection, mache etwas damit, kannst es auch löschen, und danach wird das ja nicht mehr beim Iterieren gebraucht also stört das löschen nicht. Schade.